.github/workflows/build-domain.yaml

```diff
@@ -0,0 +1,292 @@
+          }
+
+          function required_file {
+            [ -f "$1" ] || { printf '::error title=Missing file::%s\n' "$1"; exit 1; }
+          }
+
+          find domains -mindepth 1 -maxdepth 1 -type d | while read domain; do
+            required_folder "$domain/.devcontainer"
+            required_file "$domain/.devcontainer/Dockerfile"
+            required_file "$domain/.devcontainer/devcontainer.json"
+
+            required_folder "$domain/.vscode"
+            required_file "$domain/.vscode/settings.json"
+            required_file "$domain/.vscode/extensions.json"
+
+            required_file "$domain"/*.sln
+
+            required_folder "$domain/chart"
+            required_file "$domain/chart/Chart.yaml"
+            required_file "$domain/chart/values.yaml"
+
+            find "$domain" -mindepth 3 -maxdepth 3 -name "*.csproj" -exec dirname {} \; | egrep -v '[^\/]*/[^\/]*/Shared/[^\/]*' | egrep -v 'Tests$' | while read project; do
+              location=$(dirname "$project")
+              required_file "$location/chart-overrides.yaml"
+            done
+
+          done
+
+      - name: Ensure main branch is known
+        shell: bash
+        run: |
+          git fetch
+          git checkout main
+          git checkout -
+
+      - name: Find relevant domains
+        id: define-matrices
+        shell: bash
+        env:
+          REF: ${{ github.ref }}
+        run: |
+          pattern=$(find domains -mindepth 1 -maxdepth 1 -type d -exec printf '^%s|' {} \; | sed 's/|$//g')
+
+          echo 'Search pattern:'
+          echo "$pattern"
+
+          echo '::group::Change set'
+          if [ "$REF" = "refs/heads/main" ]; then
+            git show --stat=10000 main
+          else
+            git diff --stat=10000 main
+          fi
+          echo '::endgroup::'
+
+          set +e
+          if [ "$REF" = "refs/heads/main" ]; then
+            pattern=$(git show --stat=10000 main | egrep '^ [^ ]' | grep '|' | sed 's/^ *//g' | egrep -o "$pattern" | sort -u | xargs -r printf '^%s|' | sed 's/|$//g')
+          else
+            pattern=$(git diff --stat=10000 main | grep '|' | sed 's/^ *//g' | egrep -o "$pattern" | sort -u | xargs -r printf '^%s|' | sed 's/|$//g')
+          fi
+          set -e
+
+          echo 'Relevance pattern:'
+          echo "$pattern"
+
+          if [ -z "$pattern" ]; then
+            echo 'Relevance pattern is empty, emitting empty matrices.'
+            echo '::set-output name=domains::[]'
+            echo '::set-output name=projects::[]'
+            exit 0
+          fi
+
+          matrix=$(find domains -mindepth 1 -maxdepth 1 -type d | egrep "$pattern" | while read domain; do
+            projects=$(find "$domain" -mindepth 3 -maxdepth 3 -name "*.csproj" -exec dirname {} \; | egrep -v '[^\/]*/[^\/]*/Shared/[^\/]*' | egrep -v 'Tests$' | while read directory; do printf '"%s"' "$directory"; done | jq -sc '.')
```

Comment:
(Hope that the following makes sense and I am using the correct terms regarding "service" and "project")
Maybe this is something to be done at a later point, but what if we have a "service" that is only tests? An example could be BDD/Specification tests (which Team Atlas is looking into right now). These kinds of tests are not extra projects to a main project for a service - that is that it is not a test of Service A or Service B, but on a higher level.

For those kinds of tests, I would expect them to be executed on every change within the domain.

Reply:
We could include a dummy project to ensure the tests runs, but that would also publish a dummy service.

Is there a reason why these sets of tests cannot be located within an existing project? (Since that would make them run with every change within the domain as-is)